app/src/main/java/org/glucosio/android/activity/ExternalLinkActivity.java

```diff
@@ -0,0 +1,134 @@
+import java.util.HashMap;
+import java.util.Map;
+import org.glucosio.android.R;
+import org.glucosio.android.presenter.ExternalViewPresenter;
+import org.glucosio.android.tools.network.BasicNetworkConnectivity;
+import org.glucosio.android.tools.network.GlucosioExternalLinks;
+
+public class ExternalLinkActivity extends AppCompatActivity implements ExternalViewPresenter.View {
+
+    private ExternalViewPresenter presenter;
+    private WebView webView;
+    private Map<String, Integer> toolbarTitle;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_licence);
+        webView = (WebView) findViewById(R.id.webview_licence);
+        init();
+    }
+
+    private void init() {
+        initTitles();
+        initPresenter();
+        initView();
+    }
+
+    private void initView() {
+        webView.setWebViewClient(new WebViewClient() {
+            public boolean shouldOverrideUrlLoading(WebView view, WebResourceRequest request) {
+                return true;
+            }
+
+        });
+        setupToolbar();
+    }
+
+    private void initPresenter() {
+        presenter = new ExternalViewPresenter(this, new BasicNetworkConnectivity(this));
+        presenter.onViewCreated();
+    }
+
+    @Override
+    public void setupToolbarTitle(String link) {
+        if (toolbarTitle.containsKey(link)) {
+            setToolbarTitle(getString(toolbarTitle.get(link)));
+        }
+    }
+
+    @Override
+    public String extractAction() {
+        Bundle p;
+        if (getIntent().getExtras() != null) {
+            p = getIntent().getExtras();
+            return p.getString("key");
+        }
+        return null;
+    }
+
+    @Override
+    public void loadExternalUrl(String url) {
+        webView.loadUrl(url);
+    }
+
+    @Override
+    public void showNoConnectionWarning() {
+        AlertDialog dialog = new AlertDialog.Builder(this)
+            .setMessage(R.string.warning_internet_connection_required)
+            .setPositiveButton(R.string.mdtp_ok, new DialogInterface.OnClickListener() {
+                @Override public void onClick(DialogInterface dialog, int which) {
+                    ExternalLinkActivity.this.finish();
+                }
+            })
+            .create();
+        dialog.show();
+    }
+
+    private void setupToolbar() {
+        if (getSupportActionBar() != null) {
+            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+            getSupportActionBar().setDisplayShowHomeEnabled(true);
+        }
+    }
+
+    private void initTitles() {
+        toolbarTitle = new HashMap<>();
+        toolbarTitle.put(GlucosioExternalLinks.PRIVACY, R.string.preferences_privacy);
+        toolbarTitle.put(GlucosioExternalLinks.LICENSES, R.string.preferences_licences_open);
+        toolbarTitle.put(GlucosioExternalLinks.TERMS, R.string.preferences_licences_open);
```

Comment:
Nop, the activity is used to load any kind of external links. I didn't change behavior, just refactored.